### PR TITLE
Adds "hardcore mode" that, when enabled, makes starvation kill

### PR DESCRIPTION
**Pull request by unid15**
Dedicated to #6789 .
- Hardcore mode may be toggled in the secrets menu, in the _"fun"_ section. Requires FUN admin flags to enable and disable
- When enabled, player-controlled humans with less than 80 nutrition start receiving negative effects, ranging from occasional weakness and oxygen damage to seizures and, eventually, death. Also, oxygen damage is healed MUCH slower (20% of default rate) when starving
- The negative effects are:
- 60-45: You occasionally receive a message notifying you that you're REALLY hungry and going to starve soon
- 45-20: You are occasionally weakened, your eyes are blurry and you sometimes receive minor (1-15) oxy damage
- 5-20: Seizures, weakening, eye blurriness and occasional 1-25 oxygen damage. Starvation-induced seizures are half as potent as normal seizures.
- 0-5: Rapid oxygen, toxin and brain damage.

Nutrition thresholds for various starvation stages, oxy damage heal reduction and a few other things use defines, and therefore are easily configurable. Only player-controlled characters are affected!

Enabling or disabling hardcore mode will produce a global message visible to all players in world. In addition, all humans spawned during hardcore mode are informed about their need to eat!

Bonus feature: while sleeping, nutrition is decreased 75% slower.

### CODE REVIEW

**Comment by gbasood** [comment]
Is there a way for a medical team to treat someone who's critically malnourished?

**Comment by unid15** [comment]
Yeah, feeding them sugar or nutriment pills will Fix them

**Comment by ghost** [comment]
Cannibalism when.

**Comment by PJB3005** [comment]
@TermedClepe there already is canibalism...

**Comment by NigglyWiggly** [comment]
Can we also make hunger go a lot slower when this is toggled?
Starving to death after 20 minutes of not eating is pretty silly

**Comment by Ephx1** [comment]
What about a thirst stat? WHERE IS IT?

**Comment by PJB3005** [comment]
Let's take bets on the first badmin to turn this on when it gets mered.

I bet Celt.

**Comment by unid15** [comment]
@NigglyWiggly eating isn't that time consuming to be fair, 10-20 seconds is usually enough to grab and eat a few packs of crisps and forget about eating for the next 20 minutes

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -132,6 +132,15 @@
 	if(!delay_ready_dna)
 		dna.ready_dna(src)
 
+	if(hardcore_mode_on)
+		spawn(2 SECONDS)
+			//Hardcore mode stuff
+			//Warn the player that not eating will lead to his death
+			if(eligible_for_hardcore_mode(src))
+				to_chat(src, "<h5><span class='notice'>Hardcode mode is enabled!</span></h5>")
```
Comment (Zth--) [inline code/modules/mob/living/carbon/human/human.dm]: "hardcode", heh.

**Comment by ghost** [comment]
@PJB3005 

https://github.com/d3athrow/vgstation13/issues/6787

**Comment by ghost** [comment]
Heck yes.

**Comment by AutismSpider** [comment]
@NigglyWiggly takes like 5secs at most to eat any thing you can mash down the  page down key to eat and drink at sonic speeds with no risk same with getting food bustanuts are free and pizza is cheap as dirt and any one can make ice-cream for sugar or nutriment

**Pull request merged**